### PR TITLE
Patch for linux

### PR DESCRIPTION
**Pull request by benmcmahon100**
This is the accompanying change for the pr https://github.com/TanninOne/modorganizer-esptk/pull/1

### CODE REVIEW

**Comment by TanninOne** [comment]
The logic in these ifdefs is a bit broken in both these diffs.
When you want to check for windows you do `#if defined(_WIN32) || defined(_WIN64)` meaning: the block is included if either of these is defined. Which is correct.
However in other places you use `#if !defined(_WIN32) || !defined(_WIN64)` meaning: the block is included if _either_ is not defined.
This means that on 32bit windows systems you will be including the windows _and_ the "not windows" blocks because _WIN64 isn't defined.

Please use just "_WIN32", it will be defined on both 32bit and 64bit systems so you have only one macro to check for.

And then ideally please always use the same condition and have the condition as simple as possible.
Meaning: always do
#if defined(_WIN32)
// windows block
#else
// unix block
#endif

never do "#if !defined..." unless you actually have to.

This makes things easier to read.

**Comment by benmcmahon100** [comment]
Yep no that all makes sense to me! I initially started patching these at like 2am after spending a day trying to get wine to behave. I'll do the same to my other pr also.

**Comment by benmcmahon100** [comment]
Just applied the same style of changes as the other pr

**Comment by TanninOne** [comment]
Thank you for the contribution!

File: index.cpp [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
 #include <string>
+#if defined(_WIN32)
+#include "string_cast.h"
 #include <filesystem>
+namespace fs = std::filesystem;
+
+const std::wstring exstring = L".tmp";
+#else
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
```
Comment (sentriz) [inline index.cpp]: this is stupid

File: index.cpp [hunk 2]
```diff
@@ -36,9 +55,9 @@ class ESPFile {
     {
       ESP::File file(m_FileName);
       file.setLight(enable);
-      file.write(m_FileName + L".tmp");
+      file.write(m_FileName + exstring);
     }
-    std::filesystem::rename(m_FileName + L".tmp", m_FileName);
+    fs::rename(m_FileName + exstring, m_FileName);
```
Comment (sentriz) [inline index.cpp]: this is also vyr tupid

**Comment by sentriz** [comment]
appolgise wrong repository

**Pull request merged**